src/tests/functional/plugin/cpu/single_layer_tests/rnn_sequence.cpp

```diff
@@ -42,72 +71,61 @@ class RNNSequenceCPUTest : public testing::WithParamInterface<RNNSequenceCpuSpec
 
 protected:
     void SetUp() override {
-        LayerTestsDefinitions::RNNSequenceParams basicParamsSet;
-        CPUSpecificParams cpuParams;
-        std::map<std::string, std::string> additionalConfig;
-
-        size_t seq_lengths;
-        size_t batch;
-        size_t hidden_size;
-        size_t input_size;
+        std::vector<InputShape> inputShapes;
+        ngraph::helpers::SequenceTestsMode seqMode;
         std::vector<std::string> activations;
-        std::vector<float> activations_alpha;
-        std::vector<float> activations_beta;
         float clip;
-        ngraph::op::RecurrentSequenceDirection direction;
-        InferenceEngine::Precision netPrecision;
+        ov::op::RecurrentSequenceDirection direction;
+        ElementType netPrecision;
+        CPUSpecificParams cpuParams;
+        std::map<std::string, std::string> additionalConfig;
 
-        std::tie(basicParamsSet, cpuParams, additionalConfig) = this->GetParam();
+        std::tie(inputShapes, seqMode, activations, clip, direction, netPrecision, cpuParams, additionalConfig) = this->GetParam();
         std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
-        std::tie(m_mode, seq_lengths, batch, hidden_size, input_size, activations, clip, direction, netPrecision, targetDevice) = basicParamsSet;
-
-        size_t num_directions = direction == ngraph::op::RecurrentSequenceDirection::BIDIRECTIONAL ? 2 : 1;
-        std::vector<std::vector<size_t>> inputShapes = {
-            {{batch, seq_lengths, input_size},
-             {batch, num_directions, hidden_size},
-             {batch},
-             {num_directions, hidden_size, input_size},
-             {num_directions, hidden_size, hidden_size},
-             {num_directions, hidden_size}},
-        };
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+
+        init_input_shapes(inputShapes);
+
+        const size_t hiddenSize = inputDynamicShapes[1][2].get_length();
+        const size_t inputSize = inputDynamicShapes.front()[2].get_length();
+        const size_t numDirections = direction == ov::op::RecurrentSequenceDirection::BIDIRECTIONAL ? 2 : 1;
 
         configuration.insert(additionalConfig.begin(), additionalConfig.end());
 
-        if (additionalConfig[PluginConfigParams::KEY_ENFORCE_BF16] == PluginConfigParams::YES) {
-            inPrc = outPrc = Precision::BF16;
-        } else {
-            inPrc = outPrc = netPrecision;
-        }
+//        if (additionalConfig[InferenceEngine::PluginConfigParams::KEY_ENFORCE_BF16] == InferenceEngine::PluginConfigParams::YES) {
```

Comment:
uncomment please

Reply:
That does not work properly with new tests API.

Reply:
What error did you got for bf16 RNN? It work correct for https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/cpu/single_layer_tests/convolution.cpp#L152-L160 and for https://github.com/openvinotoolkit/openvino/blob/master/src/tests/functional/plugin/cpu/single_layer_tests/convolution_backprop_data.cpp#L216-L221 and for other nodes.